WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockHeaderTableViewCell.swift

```diff
@@ -32,14 +45,14 @@ class NoteBlockHeaderTableViewCell: NoteBlockTableViewCell {
 
 
     // MARK: - Public Methods
-    func downloadGravatarWithURL(_ url: URL?) {
-        if url == gravatarURL {
+    @objc func downloadGravatar(with url: URL?) {
+        guard url != gravatarURL else {
             return
         }
 
-        let placeholderImage = Style.gravatarPlaceholderImage
-        let gravatar = url.flatMap { Gravatar($0) }
-        gravatarImageView.downloadGravatar(gravatar, placeholder: placeholderImage, animate: true)
+        if let siteIconUrl = url?.absoluteString {
+            gravatarImageView.setImageWithSiteIcon(siteIconUrl, placeholderImage: Style.gravatarPlaceholderImage)
```

Comment:
This works, but when the url is a gravatar URL instead of a site icon (e.g. comment notification), it will fetch the original unresized URL instead of the resized gravatar URL.

Reply:
Let's address this in another PR, since it's a limitation of UIImageView+Gravatar.{h, m}, and may be affecting many other spots of the app.

Merging `UIImageView+Gravatar.swift` with `UIImageView+Gravatar.{h, m}` sounds like a fun project!

Reply:
[Filed here!](https://github.com/wordpress-mobile/WordPress-iOS/issues/8235)

Reply:
I don't think it's a problem with the helper (although 👍 to merging in Swift). The problem here is that `authorAvatarURL` is sometimes a gravatar and sometimes a site icon/blavatar. My take on it:

- Rename `downloadGravatar(with:)` to `downloadAuthorAvatar(with:)` (and other instances of `gravatar` with `authorAvatar`)
- Check if the url can be turned into a `Gravatar` and use `downloadGravatar`. Otherwise, treat it as a site icon, and use `setImageWithSiteIcon`

```swift
@objc(downloadAuthorAvatarWithURL:)
func downloadAuthorAvatar(with url: URL?) {
    guard url != authorAvatarURL else {
        return
    }

    authorAvatarURL = url

    guard let url = url else {
        authorAvatarImageView.image = Style.gravatarPlaceholderImage
        return
    }

    if let gravatar = Gravatar(url) {
        authorAvatarImageView.downloadGravatar(gravatar, placeholder: Style.gravatarPlaceholderImage, animate: true)
    } else {
        authorAvatarImageView.setImageWithSiteIcon(url.absoluteString)
    }
}
```